base-system/usrroot/usr/lib/hsync/libhapply.so

```diff
@@ -404,3 +405,22 @@ set_wallpaper() {
 	log "-Couldn't find any background to restore"
 	return 1 # error
 }
+
+set_scheduled_shutdown() {
```

Comment:
Let's try to make variables `local` instead of global. 
It could be something like
`local SHUTDOWN_TIME SHUTDOWN_TIME_EPOCH CURRENT_TIME_EPOCH SECONDS_TO_SHUTDOWN MINUTES_TO_SHUTDOWN`

Reply:
Done